demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+                              size_t bytesToSend )
+{
+    int32_t bytesSend = 0;
+
+    bytesSend = send( ( int ) pContext->tcpSocket, pBuffer, bytesToSend, 0 );
+
+    if( bytesSend < 0 )
+    {
+        /* Check if it was time out */
+        if( ( errno == EAGAIN ) || ( errno == EWOULDBLOCK ) )
+        {
+            /* Set return value to 0 to indicate that send was timed out. */
+            bytesSend = 0;
+        }
+    }
+
+    return bytesSend;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRecv )
+{
+    int32_t bytesReceived = 0;
+
+    bytesReceived = ( int32_t ) recv( ( int ) pContext->tcpSocket, pBuffer, bytesToRecv, 0 );
+
+    if( bytesReceived == 0 )
+    {
+        /* Server closed the connection, treat it as an error. */
+        bytesReceived = -1;
+    }
+    else if( bytesReceived < 0 )
+    {
+        /* Check if it was time out */
+        if( ( errno == EAGAIN ) || ( errno == EWOULDBLOCK ) )
+        {
+            /* Set return value to 0 to indicate nothing to receive */
+            bytesReceived = 0;
+        }
+    }
+    else
+    {
+        /* EMPTY else */
+    }
+
+    return bytesReceived;
+}
+
+/*-----------------------------------------------------------*/
+
+static int sendHttpRequest( HTTPTransportInterface_t * pTransportInterface,
+                            const char * pHost,
+                            const char * pMethod,
+                            const char * pPath )
+{
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+    HTTPRequestHeaders_t requestHeaders = { 0 };
+    HTTPRequestInfo_t requestInfo = { 0 };
+    HTTPResponse_t response = { 0 };
+
+    assert( pTransportInterface != NULL );
+    assert( pMethod != NULL );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = strlen( pHost );
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = strlen( pMethod );
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                      &requestInfo );
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Initialize the response object. The same buffer used for storing
+         * request headers is reused here. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        LogInfo( ( "Sending HTTP %s request to %s%s...",
```

Comment:
For the server host and path combination, should it be formatted as "%s/%s"?

Reply:
No, the path already includes the leading /